python/mlx/nn/layers/positional_encoding.py

```diff
@@ -104,9 +104,9 @@ def create_cos_sin_theta(
         dtype=mx.float32,
     ):
         if (N, D, offset, base, scale, dtype) != cls._cos_sin_theta_key:
-            D = D // 2
+            half_D = D // 2
             positions = mx.arange(offset, N, dtype=dtype) * scale
-            freqs = mx.exp(-mx.arange(0.0, D, dtype=dtype) * (math.log(base) / D))
+            freqs = mx.exp(-mx.arange(0.0, half_D, dtype=dtype) * (math.log(base) / half_D))
             theta = mx.reshape(positions, (-1, 1)) * mx.reshape(freqs, (1, -1))
             cls._cos_sin_theta_key = (N, D, offset, base, scale, dtype)
```

Comment:
Note that originally it was using `D // 2` as the cache key here

Reply:
Nice catch! Thanks.